src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs

```diff
@@ -0,0 +1,208 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.WindowsRuntime.Internal;
+using System.Threading;
+using System.Collections.ObjectModel;
+using System.Security;
+using System.Threading.Tasks;
+
+namespace System.IO
+{
+    /// <summary>
+    /// A BufferedStream wrapper which expose BufferedStream's UnderlyingStream and BufferSize
+    /// </summary>
+    internal sealed class BufferedStreamWrapper : Stream
+    {
+        private BufferedStream _bufferedStream;
+        private Stream _stream;                               // Underlying stream.  Close sets _stream to null.
+        private readonly Int32 _bufferSize;                   // Length of internal buffer (not counting the shadow buffer).
```

Comment:
Nit: The corefx coding style is to use language keywords instead of BCL types, so `Int32`, `Int64`, `Byte`, `String`, etc. should be `int`, `long`, `byte`, `string`, etc. Applies throughout.